Storage/container/many_to_many.h

```diff
@@ -57,51 +58,48 @@ class GenericMatrix {
   using DEPRECATED_T_(ROW) = row_t;
   using DEPRECATED_T_(COL) = col_t;
 
-  explicit GenericMatrix(MutationJournal& journal) : journal_(journal) {}
+  explicit GenericManyToMany(MutationJournal& journal) : journal_(journal) {}
 
   bool Empty() const { return map_.empty(); }
   size_t Size() const { return map_.size(); }
 
   void Add(const T& object) {
     const auto row = sfinae::GetRow(object);
     const auto col = sfinae::GetCol(object);
-    const auto row_col = std::make_pair(row, col);
-    const auto it = map_.find(row_col);
-    if (it != map_.end()) {
-      const T previous_object = *(it->second);
+    const auto key = std::make_pair(row, col);
+    const auto cit = map_.find(key);
+    if (cit != map_.end()) {
+      const T& previous_object = *(cit->second);
       journal_.LogMutation(UPDATE_EVENT(object),
-                           [this, row_col, previous_object]() { DoAdd(row_col, previous_object); });
+                           [this, key, previous_object]() { DoAdd(key, previous_object); });
     } else {
-      journal_.LogMutation(UPDATE_EVENT(object), [this, row_col]() { DoErase(row_col); });
+      journal_.LogMutation(UPDATE_EVENT(object), [this, key]() { DoErase(key); });
     }
-    DoAdd(row_col, object);
+    DoAdd(key, object);
   }
 
-  void Erase(sfinae::CF<row_t> row, sfinae::CF<col_t> col) {
-    const auto row_col = std::make_pair(row, col);
-    const auto it = map_.find(row_col);
-    if (it != map_.end()) {
-      const T previous_object = *(it->second);
+  void Erase(const key_t& key) {
+    const auto cit = map_.find(key);
+    if (cit != map_.end()) {
+      const T& previous_object = *(cit->second);
       journal_.LogMutation(DELETE_EVENT(previous_object),
-                           [this, row, col, row_col, previous_object]() {
-                             auto& placeholder = map_[row_col];
-                             placeholder = std::make_unique<T>(previous_object);
-                             forward_[row][col] = placeholder.get();
-                             transposed_[col][row] = placeholder.get();
-                           });
-      DoErase(row_col);
+                           [this, key, previous_object]() { DoAdd(key, previous_object); });
+      DoErase(key);
     }
   }
-  void Erase(const std::pair<row_t, col_t>& key) { Erase(key.first, key.second); }
+  void Erase(sfinae::CF<row_t> row, sfinae::CF<col_t> col) { Erase(std::make_pair(row, col)); }
```

Comment:
You're right. If there's a `std::pair<row_t, col_t>`as a key in `std::[unordered_]map`, the `Erase(row, col)` method with two signatures would inevitably create that `std::pair<row_t, col_t>` on the stack.

Good point. Confusion resolved. :-)